server/jetstream_cluster_test.go

```diff
@@ -11206,6 +11206,71 @@ func TestJetStreamRemovedPeersAndStreamsListAndDelete(t *testing.T) {
 	require_Error(t, err, nats.ErrStreamNotFound)
 }
 
+func TestJetStreamConsumerDeliverNewBug(t *testing.T) {
+	sc := createJetStreamSuperCluster(t, 3, 3)
+	defer sc.shutdown()
+
+	pcn := "C2"
+	sc.waitOnLeader()
+	ml := sc.leader()
+	if ml.ClusterName() == pcn {
+		pcn = "C1"
+	}
+
+	// Client based API
+	nc, js := jsClientConnect(t, ml)
+	defer nc.Close()
+
+	_, err := js.AddStream(&nats.StreamConfig{
+		Name:      "T",
+		Replicas:  3,
+		Placement: &nats.Placement{Cluster: pcn},
+	})
+	require_NoError(t, err)
+
+	// Put messages in..
+	num := 200
+	for i := 0; i < num; i++ {
+		js.PublishAsync("T", []byte("OK"))
+	}
+	select {
+	case <-js.PublishAsyncComplete():
+	case <-time.After(5 * time.Second):
+		t.Fatalf("Did not receive completion signal")
+	}
+
+	ci, err := js.AddConsumer("T", &nats.ConsumerConfig{
+		Durable:       "d",
+		AckPolicy:     nats.AckExplicitPolicy,
+		DeliverPolicy: nats.DeliverNewPolicy,
+	})
+	require_NoError(t, err)
+
+	if ci.Delivered.Consumer != 0 || ci.Delivered.Stream != 200 {
+		t.Fatalf("Incorrect consumer delivered info: %+v", ci.Delivered)
+	}
+
+	c := sc.clusterForName(pcn)
+	for _, s := range c.servers {
+		sd := s.JetStreamConfig().StoreDir
+		s.Shutdown()
+		removeDir(t, sd)
+		s = c.restartServer(s)
+		c.waitOnServerHealthz(s)
+	}
+
+	c.waitOnConsumerLeader("$G", "T", "d")
```

Comment:
the way I read your email, don't these two lines mask the issue?
The issue happened before the consumer leader was ready, but didn't happen after.
So shouldn't one instance of the test avoid these and a much shorter test test that once `c.waitOnServerHealthz(z)` returned the consumer has a leader (vs wait for one)

Reply:
Will merge but we can add on additional torture tests for state being wiped or corrupt for sure.